docker/docker-compose.yml

```diff
@@ -42,32 +31,40 @@ mesosslave1:
   - MESOS_CONTAINERIZERS=docker,mesos
   - DOCKER_DAEMON_ARGS="--log-level=error"
   links:
-  - etcd
   - mesosmaster1
-  - "ambassador:apiserver"
   volumes:
   - /var/run/docker.sock:/var/run/docker.sock
+  - /sys:/sys
+  - /usr/local/bin/docker:/usr/bin/docker
```

Comment:
I have split the dind and non-dind variant. Here the second one is used, i.e. no wrapdocker and hence no mounted cgroups. That's why /sys is mounted.

The problem with installing docker is that docker has changed the API version multiple times. Then your inner docker cannot speak with the outer anymore. This way it can. On the other hand, the outer binary must be at the specific location in the filesystem and be runnable in the container. No free lunch with either way.